### DISCUSSION TITLE
Opt-in models on search or ability for plugins to add searchable models

### DISCUSSION THREAD

**Discussion by esoubihe** [netbox-community/netbox#7936] [Ideas]
Users might have more data in models that are currently not searchable. E.g. One might want to use Tenant Groups for their customer names and Tenants for more granular organization, such as the customer project.

Currently, there's no way to search for models that are not explicitly added to search. For these use cases, it could be interesting to a) allow opt-in for additional models on the configuration or b) allow plugins to add different models to the global search page.

Examples of this can range for the example mentioned above and other plugins, such as [BGP](https://github.com/k01ek/netbox-bgp), which could have lots of data.

**Comment by PieterL75** [comment]
I would also like to be able to search 'Custom Fields' using the top-search-bar.
currently, you have to go to the object and do a specific search
